packages/ui/src/internalComponents/Popup/index.tsx

```diff
@@ -291,6 +293,40 @@ export const Popup = forwardRef(
       [unmountTooltip],
     )
 
+    // Handle hide on esc press
+    useEffect(() => {
+      const handleEscPress = (event: KeyboardEvent) => {
+        if (event.key === 'Escape') {
+          event.preventDefault()
+          event.stopPropagation()
+          debounceTimer.current = setTimeout(() => {
+            setReverseAnimation(true)
+            timer.current = setTimeout(() => {
+              unmountTooltip()
+              onClose?.()
+            }, ANIMATION_DURATION)
+          }, 200)
```

Comment:
Could we avoid duplicating this whole block (lines 227-233)?

Reply:
@QuiiBz fixed!